modules/data.atmosphere/R/metgapfill.R

```diff
@@ -302,10 +302,12 @@ metgapfill <- function(in.path, in.prefix, outfolder, start_date, end_date, lst
 
     ##Once all are filled, do one more consistency check
     es <- get.es(Tair_degC) * 100
+
     rH[rH < 0] <- 0
     rH[rH > 100] <- 100
     VPD[VPD < 0] <- 0
-    VPD[VPD > es] <- es
+    badVPD_es <- which(VPD > es)
+    VPD[badVPD_es] <- es[badVPD_es]
```

Comment:
If you're using which just for the purpose of excluding NAs, you could consider doing the exclusion explicitly: `VPD[VPD > es & !is.na(VPD)] <- es`.

Reply:
Your solution doesn't work, VPD>es & !is.na(VPD) will return TRUE/FALSE/NA, and since es is a vector of more than one element, will trigger the same error. I tested this explicitly.

Reply:
Gotcha -- I missed that es had more than one element. Sorry for the noise.